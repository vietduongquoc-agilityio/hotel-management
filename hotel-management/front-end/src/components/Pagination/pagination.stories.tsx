import { useState } from "react";
import { Meta, StoryFn } from "@storybook/react";
import Pagination from ".";

export default {
  title: "Components/Pagination",
  component: Pagination,
  argTypes: {
    currentPage: {
      control: { type: "number" },
      description: "The current page being displayed",
    },
    pageCount: {
      control: { type: "number" },
      description: "Total number of pages",
    },
    pageSize: {
      control: { type: "number" },
      description: "Number of items per page",
    },
  },
} as Meta;

const Template: StoryFn = (args) => {
  const [currentPage, setCurrentPage] = useState(args.currentPage);

  return (
    <Pagination
      pageSize={15}
      pageCount={2}
      {...args}
      currentPage={currentPage}
      setCurrentPage={setCurrentPage}
    />
  );
};

export const Default = Template.bind({});
Default.args = {
  currentPage: 1,
  pageCount: 5,
  pageSize: 10,
};

export const MidPage = Template.bind({});
MidPage.args = {
  currentPage: 3,
  pageCount: 10,
  pageSize: 10,
};

export const LastPage = Template.bind({});
LastPage.args = {
  currentPage: 10,
  pageCount: 10,
  pageSize: 10,
};

export const SinglePage = Template.bind({});
SinglePage.args = {
  currentPage: 1,
  pageCount: 1,
  pageSize: 10,
};
