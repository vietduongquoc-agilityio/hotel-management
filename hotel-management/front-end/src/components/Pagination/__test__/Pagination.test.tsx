import { render, screen, fireEvent } from "@testing-library/react";
import Pagination from "../index";

describe("Pagination Component", () => {
  const mockSetCurrentPage = jest.fn();

  const defaultProps = {
    currentPage: 1,
    setCurrentPage: mockSetCurrentPage,
    pageSize: 5,
    pageCount: 5,
  };

  beforeEach(() => {
    mockSetCurrentPage.mockClear();
  });

  it("renders correctly", () => {
    render(<Pagination {...defaultProps} />);
    expect(screen.getByText("< Previous")).toBeInTheDocument();
    expect(screen.getByText("Next >")).toBeInTheDocument();
  });

  it("disables the 'Previous' button on the first page", () => {
    render(<Pagination {...defaultProps} />);
    const previousButton = screen.getByText("< Previous");
    expect(previousButton).toBeDisabled();
  });

  it("disables the 'Next' button on the last page", () => {
    render(<Pagination {...defaultProps} currentPage={defaultProps.pageCount} />);
    const nextButton = screen.getByText("Next >");
    expect(nextButton).toBeDisabled();
  });

  it("renders the correct number of page buttons", () => {
    render(<Pagination {...defaultProps} />);
    const pageButtons = screen.getAllByRole("button", { name: /^\d+$/ });
    expect(pageButtons.length).toBe(defaultProps.pageCount);
    pageButtons.forEach((button, index) => {
      expect(button).toHaveTextContent(`${index + 1}`);
    });
  });

  it("calls setCurrentPage with the correct page number when a page button is clicked", () => {
    render(<Pagination {...defaultProps} />);
    const pageButton = screen.getByText("3");
    fireEvent.click(pageButton);
    expect(mockSetCurrentPage).toHaveBeenCalledWith(3);
  });

  it("calls setCurrentPage with the previous page when 'Previous' is clicked", () => {
    render(<Pagination {...defaultProps} currentPage={3} />);
    const previousButton = screen.getByText("< Previous");
    fireEvent.click(previousButton);
    expect(mockSetCurrentPage).toHaveBeenCalledWith(2);
  });

  it("calls setCurrentPage with the next page when 'Next' is clicked", () => {
    render(<Pagination {...defaultProps} currentPage={3} />);
    const nextButton = screen.getByText("Next >");
    fireEvent.click(nextButton);
    expect(mockSetCurrentPage).toHaveBeenCalledWith(4);
  });
});
