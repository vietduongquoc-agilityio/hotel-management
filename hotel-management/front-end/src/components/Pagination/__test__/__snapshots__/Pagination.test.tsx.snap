// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Component renders correctly 1`] = `
<div>
  <div
    class="chakra-stack css-r3l8tq"
  >
    <div
      class="chakra-stack css-1r1rpqn"
    >
      <button
        class="chakra-button css-57a4je"
        secondary=""
        type="button"
      >
        &lt; Previous
      </button>
      <div
        class="chakra-stack css-1igwmid"
      >
        <button
          class="chakra-button css-17vol85"
          type="button"
        >
          1
        </button>
        <button
          class="chakra-button css-bdyd46"
          type="button"
        >
          2
        </button>
        <button
          class="chakra-button css-bdyd46"
          type="button"
        >
          3
        </button>
        <button
          class="chakra-button css-bdyd46"
          type="button"
        >
          4
        </button>
        <button
          class="chakra-button css-bdyd46"
          type="button"
        >
          5
        </button>
      </div>
      <button
        class="chakra-button css-57a4je"
        type="button"
      >
        Next &gt;
      </button>
    </div>
  </div>
</div>
`;
